junebug/workers.py

```diff
@@ -242,8 +262,9 @@ def request_failed(resp):
     return resp.code < 200 or resp.code >= 300
 
 
-def post(url, data):
+def post(url, data, timeout):
     return treq.post(
         url.encode('utf-8'),
         data=json.dumps(data, cls=JSONMessageEncoder),
-        headers={'Content-Type': 'application/json'})
+        headers={'Content-Type': 'application/json'},
+        timeout=timeout)
```

Comment:
If the timeout is reached, this would raise an exception. I think we should probably catch that exception and log it, rather than letting it rise up.

There are a few exceptions to catch with this: https://github.com/twisted/treq/issues/117